Lib/fontTools/ttLib/tables/_g_l_y_f.py

```diff
@@ -110,7 +112,9 @@ def compile(self, ttFont):
 			ttFont['maxp'].numGlyphs = len(self.glyphs)
 		return data
 
-	def toXML(self, writer, ttFont, progress=None):
+	def toXML(self, writer, ttFont, progress=None, tablePath=None, 
+            ttLibVersion=None, idlefunc=None, newlinestr=None, splitGlyphs=None):
```

Comment:
actually, can't we take all these extra values from the current writer instance?
- tablePath is writer.file.name
- ttLibVersion we can compute again from fonttools.version
- idlefunc is writer.idlefunc
- newlinestr is writer.newlinestr

I wouldn't like the change the API for tables' `toXML` methods. They are all predictably the same.
But I see that you need to tell the glyf table that you want to `splitGlyphs` so I see no way around adding a extra optional keyword argument `splitGlyphs=False` (not `None`)

Reply:
k. great.  Yes, I agree and I'll remove the unnecessary extra parameters and extract the values from the writer when they are needed inside of the glyp writing function.